### DISCUSSION TITLE
Shader Compile Time Mitigation

### DISCUSSION THREAD

**Discussion by i2iRobert** [alexmercerind/dart_vlc#350] [Q&A] [answered]
We're writing a Windows Flutter app, the issue we've come across with dart_vlc appears to be shader compilation time.

The example app demonstrates the issue quite clearly, as it loads first time it takes around 10 seconds to "compile" the player for the first time.

Which can be seen in this screenshot of the performance dev tools when running the app we are working on:

![dart_vlc_shader_compile_time-app](https://user-images.githubusercontent.com/107061316/209961713-a35531d2-97fa-478c-99c4-a0620801e5fc.png)

I attempted to capture the "jank" on the example app but this happens as the app is loading and the performance tools aren't opened on load.

What would be the best way to mitigate this "compile" time on windows, as you don't appear to be able to pre cache shaders on Windows?

In the example app the video player is the first screen you see, so the delay appears to be "loading", however in the app we've made the video player is not the first thing you see, so when you click for the video player there will be a solid 10 seconds of loading/jank/pause before anything happens, making for a poor user experience. This only happens the first time on a fresh build/install, once its "compiled" once it loads as fast as I would expect.

Any pointers, or suggestions are welcome.

**Comment by alexmercerind** [comment, marked as answer]
Sorry, this has nothing to do with shader pre-compilation etc. etc. It has nothing to do with Flutter either.
I have explained everything in simpler terms here: [Video: Hardware Acceleration](https://github.com/alexmercerind/dart_vlc/issues/345).

<hr>

Also, I as the author of [package:dart_vlc](https://github.com/alexmercerind/dart_vlc) don't feel satisfied for it's performance. I have started a new project [package:media_kit](https://github.com/alexmercerind/media_kit) which offers better, performant & stable media playback which is hardware accelerated. Please consider using it instead.

This project ([package:dart_vlc](https://github.com/alexmercerind/dart_vlc)) will also receive improvements in future, depending upon feasibility as discussed on the issue linked above.
